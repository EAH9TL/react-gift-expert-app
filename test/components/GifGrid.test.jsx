const { render, screen } = require("@testing-library/react");
const { GifGrid } = require("../../src/components/GifGrid");
const { useFetchGifs } = require("../../src/hooks/useFetchGifs");

jest.mock("../../src/hooks/useFetchGifs");

describe("Probando GifGrid", () => {
  const category = "Naruto";
  test("should mostrar loading inicialmente", () => {
    useFetchGifs.mockReturnValue({
      images: [],
      isLoading: true,
    });
    render(<GifGrid category={category} />);
    expect(screen.getByText("Cargando...")).toBeTruthy();
    expect(screen.getByText(category)).toBeTruthy();
  });

  test("should mostrar items cuando se cargan las imagenes mediante el useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        title: "Naruto",
        url: "https://localhost/naruto.jpg",
      },
      {
        id: "123",
        title: "One Piece",
        url: "https://localhost/one-piece.jpg",
      },
    ];
    useFetchGifs.mockReturnValue({
      images: gifs,
      isLoading: false,
    });
    const { container } = render(<GifGrid category={category} />);
    expect(container).toMatchSnapshot();

    expect(screen.getAllByRole("img").length).toBe(2);
  });
});
