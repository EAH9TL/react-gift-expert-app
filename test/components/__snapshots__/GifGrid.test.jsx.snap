// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GifGrid should mostrar items cuando se cargan las imagenes mediante el useFetchGifs 1`] = `
<div>
  <h3>
    Naruto
  </h3>
  <div
    class="card-grid"
  >
    <div
      class="card"
    >
      <img
        alt="Naruto"
        src="https://localhost/naruto.jpg"
      />
      <p>
        Naruto
      </p>
    </div>
    <div
      class="card"
    >
      <img
        alt="One Piece"
        src="https://localhost/one-piece.jpg"
      />
      <p>
        One Piece
      </p>
    </div>
  </div>
</div>
`;
